### ISSUE TITLE
Invalid argument supplied for foreach() in ApplicationController ($multiUploadFiles)

### ISSUE THREAD

**Issue by luii** [itx-informationssysteme/jobapplications#22]
https://github.com/itx-informationssysteme/jobapplications/blob/2a456ed2c942f228238e2fa05a7c3ddd6163db58/Classes/Controller/ApplicationController.php#L508-L514

Gives me this error

```
Oops, an error occurred!
PHP Warning: Invalid argument supplied for foreach() in /var/www/html/typo3conf/ext/jobapplications/Classes/Controller/ApplicationController.php line 508
More information regarding this error might be available online.
```

Legacy Upload is enabled:

```typoscript
[tx_jobapplications]
  [view]
    [templateRootPaths]
      [0] = EXT:{extension.shortExtensionKey}/Resources/Private/Templates/
      [1] = EXT:jobapplications/Resources/Private/Templates/
      [2] = EXT:****/Resources/Private/Job/Templates/
    [partialRootPaths]
      [0] = EXT:jobapplications/Resources/Private/Partials/
      [1] = EXT:jobapplications/Resources/Private/Partials/
      [2] = EXT:****/Resources/Private/Job/Partials/
    [layoutRootPaths]
      [0] = EXT:jobapplications/Resources/Private/Layouts/
      [1] = EXT:jobapplications/Resources/Private/Layouts/
      [2] = EXT:****/Resources/Private/Job/Layouts/
    [widget]
      [TYPO3\CMS\Fluid\ViewHelpers\Widget\PaginateViewHelper]
    [settings]
      [enableGoogleJobs] = 0
      [emailContentType] = both
      [legacy_upload] = 1
      [defaultContactMailAddress] = address@address.de
      [defaultContactFirstName] = Default First Name
      [defaultContactLastName] = Default Last Name
      [maxItemsOnPage] = 9
      [list]
      [filter]
    [persistence]
    [features]
    [mvc]
```

Still im not entirely sure why this is happening

**Comment by der-phillip** [comment]
I have the same error. Depending on if legacy_upload is switched on or off in line 499 or 508.
I hope you can fix it soon.

**Comment by luii** [comment]
@der-phillip did you've tried my quick fix #23 ? This only happens because this check is missing.
https://github.com/itx-informationssysteme/jobapplications/blob/2a456ed2c942f228238e2fa05a7c3ddd6163db58/Classes/Controller/ApplicationController.php#L403-L414

At this part it checks for multifile upload and so it should check at the lines referenced in #23 .
Until the @itx-benjamin-jasper fixes it you can use this fix and not update the Extension in TYPO3.
